### PR TITLE
Feat : crowdin translations

### PR DESCRIPTION
**Pull request by ga-devfront**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Update current translations and templates to use only module translation (not core) with Crowdin.
| Type?             | improvement
| BC breaks?        | no (or yes if you consider lost some translation is a BC)
| Deprecations?     | no
| Sponsor company   | @PrestaShopCorp
| How to test?      | The module should work as before. Check that the languages ​​work: cs, de, en, es, fr, it, pl, ru. If Translation is not translated it's possible he fall back to english.

### CODE REVIEW

**Comment by ps-jarvis** [comment]
This pull request seems to contain new translation strings. I have summarized them below to ease up review:

- `Modules.Autoupgrade.Admin`
    - [ ] `N/A`
    - [ ] `Yes`
    - [ ] `No`
    - [ ] `Delete`
    - [ ] `Available`
    - [ ] `Name:`
    - [ ] `Save`
    - [ ] `PHP Version notice`

(Note: this is an automated message, but answering it will reach a real human)

File: classes/UpgradeTools/Translator.php [hunk 1]
```diff
@@ -51,19 +61,26 @@ public function __construct($caller)
...
 
-        if (method_exists('\Translate', 'getModuleTranslation')) {
-            $translated = \Translate::getModuleTranslation('autoupgrade', $id, $this->caller, null);
-            if (!count($parameters)) {
-                return $translated;
-            }
-        } else {
-            $translated = $id;
+        // Use new translation system for PrestaShop 1.7.6 and newer
+        if (version_compare($psVersion, '1.7.6.0', '>=')) {
```
Comment (Hlavtox) [inline classes/UpgradeTools/Translator.php:73]: I did not test this guys, but I think that you need to targe 1.7.8 instead. 🤔

1.7.8 is the first version of Prestashop, that could load translations from `modules/modulename/translations` folder, on 1.7.6 and 1.7.7, they need to be imported manually into database.

**Comment by Hlavtox** [comment]
@Quetzacoalt91 @Julievrz Guys, is it possible to fix the repo name on crowdin? There is some branch name or something.
Or is it possible to change it later? I don't want to start translating it and then it being deleted. ;-)

Snímek obrazovky 2024-06-19 095240

**Comment by ga-devfront** [comment]
> @Quetzacoalt91 @Julievrz Guys, is it possible to fix the repo name on crowdin? There is some branch name or something. Or is it possible to change it later? I don't want to start translating it and then it being deleted. ;-)
> 
> Snímek obrazovky 2024-06-19 095240

It's normal for the moment, you shouldn't start translating. You will have to wait for this PR to merge before changing the branch name to connect to Crowdin.

**Comment by Hlavtox** [comment]
@ga-devfront Ok, will wait until you say it's ready. ;-)

File: autoupgrade.php [hunk 2]
```diff
@@ -121,26 +126,6 @@ public function uninstall()
...
     }
 
-    /**
-     * Register the current module to a given hook and moves it at the first position.
-     *
-     * @param string $hookName
-     *
-     * @return bool
-     */
-    public function registerHookAndSetToTop($hookName)
```
Comment (Quetzacoalt91) [inline autoupgrade.php:131]: Can you please provide an upgrade file to unregister the hook?

File: composer.lock [hunk 3]
```diff
@@ -1,3353 +0,0 @@
...
-            "description": "Assertions to validate method input/output with nice error messages.",
-            "keywords": [
-                "assert",
-                "check",
-                "validate"
-            ],
-            "support": {
-                "issues": "https://github.com/webmozarts/assert/issues",
-                "source": "https://github.com/webmozarts/assert/tree/1.9.1"
-            },
```
Comment (Quetzacoalt91) [inline composer.lock:3335]: This file should not be deleted

File: classes/UpgradeTools/Translator.php [hunk 4]
```diff
@@ -1,67 +1,66 @@
...
+        if (file_exists($path)) {
+            $this->loadXlfFile($path);
+        }
     }
 
     /**
-     * Translate a string to the current language.
+     * Load translations from a specific XLF file.
+     *
+     * @param string $filePath path to the XLF file
```
Comment (M0rgan01) [inline classes/UpgradeTools/Translator.php:31]:
```suggestion
     * Load translations from a specific XLF file.
     *
     * @return void
     * @param string $filePath path to the XLF file
```

File: classes/UpgradeTools/Translator.php [hunk 5]
```diff
@@ -91,4 +90,9 @@ public function applyParameters($id, array $parameters = [])
 
         return call_user_func_array('sprintf', array_merge([$id], $parameters));
     }
+
+    public function getLocale()
```
Comment (M0rgan01) [inline classes/UpgradeTools/Translator.php:94]: Add php doc for return

File: classes/UpgradeTools/Translator.php [hunk 6]
```diff
@@ -1,67 +1,66 @@
...
- */
-
 namespace PrestaShop\Module\AutoUpgrade\UpgradeTools;
 
+use SimpleXMLElement;
+
 class Translator
 {
-    private $caller;
+    private $translations = [];
```
Comment (M0rgan01) [inline classes/UpgradeTools/Translator.php:9]: Add phpdoc for type

File: classes/UpgradeTools/Translator.php [hunk 7]
```diff
@@ -1,67 +1,66 @@
...
 
-    public function __construct($caller)
+    /**
+     * Load translations from XLF files.
+     */
```
Comment (M0rgan01) [inline classes/UpgradeTools/Translator.php:13]:
```suggestion
    /**
     * Load translations from XLF files.
     * @return void
     */
```

File: classes/UpgradeTools/Translator.php [hunk 8]
```diff
@@ -1,67 +1,66 @@
...
-        if (method_exists('\Translate', 'getModuleTranslation')) {
-            $translated = \Translate::getModuleTranslation('autoupgrade', $id, $this->caller, null);
-            if (!count($parameters)) {
-                return $translated;
-            }
-        } else {
-            $translated = $id;
+        if (empty($this->translations)) {
+            $this->loadTranslations();
         }
```
Comment (M0rgan01) [inline classes/UpgradeTools/Translator.php:62]: This may throw exceptions. What is the scenario in this case?

File: upgrade/install-5.1.0.php [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+/**
+ * Manually remove the legacy controller. It has been deleted from the project but remain present while upgrading the module.
+ *
```
Comment (M0rgan01) [inline upgrade/install-5.1.0.php:32]: Outdated

File: upgrade/install-5.1.0.php [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+ * @return bool
+ */
+function upgrade_module_5_1_0($module)
+{
+    $module->unregisterHook('dashboardZoneOne');
+}
```
Comment (M0rgan01) [inline upgrade/install-5.1.0.php:38]: 5.1.0 ? Not 6.0.0 ?

File: autoupgrade.php [hunk 11]
```diff
@@ -58,6 +58,11 @@ public function __construct()
         $this->ps_versions_compliancy = ['min' => '1.7.0.0', 'max' => _PS_VERSION_];
     }
 
+    public function isUsingNewTranslationSystem()
+    {
+        return true;
+    }
+
```
Comment (M0rgan01) [inline autoupgrade.php:65]: Who uses this function?

File: classes/TaskRunner/AbstractTask.php [hunk 12]
```diff
@@ -46,7 +47,7 @@ abstract class AbstractTask
     protected $logger;
 
     /**
-     * @var \PrestaShop\Module\AutoUpgrade\UpgradeTools\Translator
+     * @var Translator
      */
     protected $translator;
```
Comment (M0rgan01) [inline classes/TaskRunner/AbstractTask.php:53]: Be careful when creating an empty `RemoveSamples` file

File: upgrade/install-6.0.0.php [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+/**
+ * Manually remove the dashboardZoneOne hook.
+ *
+ * @return bool
```
Comment (Quetzacoalt91) [inline upgrade/install-6.0.0.php:33]: Does PHPStan yells about this ? because the method does not return anything

**Review by Quetzacoalt91 — APPROVED** [review]

**Review by M0rgan01 — APPROVED** [review]

**Comment by AureRita** [comment]
Hi @ga-devfront 

Thank you for your PR, I tested it and it seems to works as you can see : 

[recording(242).webm](https://github.com/PrestaShop/autoupgrade/assets/100699445/461eb7ee-1fbd-4f2a-b03e-71a3fb62e3e5)


Tested from : 
1.7.8.11 to 8.1.7
8.0.5 to 8.1.7
8.1.5 to 8.1.7
8.0.5 to 9.0.0 
8.1.7 to 9.0.0

FYI, When you do an upgrade from 1.7.8.11 to 8.1.7, you don't have the sidebar translate when you change the languages as you can see 

image

If the "issue" from the version 1.7.8.11 shouldn't be corrected by your PR, It's QA :heavy_check_mark: If it should be corrected by it, we have to correct it 

Waiting for your feedback

**Comment by ga-devfront** [comment]
> Hi @ga-devfront
> 
> Thank you for your PR, I tested it and it seems to works as you can see :
> 
>  recording.242.webm 
> Tested from : 1.7.8.11 to 8.1.7 8.0.5 to 8.1.7 8.1.5 to 8.1.7 8.0.5 to 9.0.0 8.1.7 to 9.0.0
> 
> FYI, When you do an upgrade from 1.7.8.11 to 8.1.7, you don't have the sidebar translate when you change the languages as you can see
> 
> image
> 
> If the "issue" from the version 1.7.8.11 shouldn't be corrected by your PR, It's QA ✔️ If it should be corrected by it, we have to correct it
> 
> Waiting for your feedback

I'm not sure you are testing the right thing. This PR takes care of the translations of the module. It is therefore necessary to test that the translations work well on the module in the following languages: cs, de, en, es, fr, it, pl, ru.

However if this is what you tested we are good and can merge!

**Review by AureRita — APPROVED** [review]
Hi @ga-devfront 

I checked them with other Page, I check it also in 1.7.1.0 as we said to see if there is no issue on it and as you can see : 

[recording(244).webm](https://github.com/PrestaShop/autoupgrade/assets/100699445/ecea8911-9fe3-4010-ad20-a2aa6318261c)

It works ! so it's a QA :heavy_check_mark: 

Thank you

**Pull request merged**